http/connection_wrapper.go

```diff
@@ -0,0 +1,133 @@
+package http
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"reflect"
+	"strings"
+
+	"github.com/arangodb/go-driver"
+	"github.com/arangodb/go-velocypack"
+)
+
+type connectionDebugWrapper struct {
```

Comment:
you can change the stucture into:
```
type connectionDebugWrapper struct {
	driver.Connection
	ct         driver.ContentType
}
```

And then you don't have to create functions like:
```
func (c *connectionDebugWrapper) NewRequest(method, path string) (driver.Request, error) {
	return c.connOrigin.NewRequest(method, path)
}
```
because that function does nothing new, it is only wrapper.
all you need to do is to have only two new functions:
```
func (c *connectionDebugWrapper) Do(ctx context.Context, req driver.Request) (driver.Response, error)
func (c *connectionDebugWrapper) Unmarshal(data driver.RawObject, result interface{}) error
```
because these functions do something new.